File: algorithm/java/fibonacci.java
public class Main {
   public static void main(String[] args) throws Exception {
       long start = System.nanoTime();
       System.out.println("result:" + fib(40));
       System.out.println("time:" + (System.nanoTime() - start) /1000000000.0 + " sec");
   }
   
   static int fib(int n){
       if (n < 2){
           return n;
       }
       return fib(n - 2) + fib(n - 1);
   }
}

File: algorithm/rust/fibonacci.rs
use std::time::{Instant};

fn main(){
    let start = Instant::now();
    println!("result: {}", fib(40));
    let end = start.elapsed();
    println!("time: {}.{:03} sec", end.as_secs(), end.subsec_nanos() / 1000000);
}

fn fib(n: i32) -> i32 {
    if n < 2 {
        return n;
    }
    return fib(n - 2) + fib(n - 1);
}